sympy/physics/continuum_mechanics/beam.py

```diff
@@ -3332,3 +3332,229 @@ def plot_shear_stress(self, dir="all", subs=None):
             Py = self._plot_shear_stress('y', subs)
             Pz = self._plot_shear_stress('z', subs)
             return PlotGrid(3, 1, Px, Py, Pz)
+
+    def max_shear_force(self, dir=None):
+        """
+        Returns point of max shear force and its corresponding deflection value
+        along the specified direction in a Beam object.
+
+        Parameters
+        ==========
+        dir : string
+               Direction along which max shear force is required.
+
+        Examples
+        ========
+        There is a beam of length 20 meters. It it supported by rollers
+        at of its end. A linear load having slope equal to 12 is applied
+        along y-axis. A constant distributed load of magnitude 15 N is
+        applied from start till its end along z-axis.
+
+        .. plot::
+            :context: close-figs
+            :format: doctest
+            :include-source: True
+
+            >>> from sympy.physics.continuum_mechanics.beam import Beam3D
+            >>> from sympy import symbols
+            >>> l, E, G, I, A, x = symbols('l, E, G, I, A, x')
+            >>> b = Beam3D(20, 40, 21, 100, 25, x)
+            >>> b.apply_load(15, start=0, order=0, dir="z")
+            >>> b.apply_load(12*x, start=0, order=0, dir="y")
+            >>> b.bc_deflection = [(0, [0, 0, 0]), (20, [0, 0, 0])]
+            >>> R1, R2, R3, R4 = symbols('R1, R2, R3, R4')
+            >>> b.apply_load(R1, start=0, order=-1, dir="z")
+            >>> b.apply_load(R2, start=20, order=-1, dir="z")
+            >>> b.apply_load(R3, start=0, order=-1, dir="y")
+            >>> b.apply_load(R4, start=20, order=-1, dir="y")
+            >>> b.solve_for_reaction_loads(R1, R2, R3, R4)
+            >>> b.solve_slope_deflection()
```

Comment:
If `b.solve_for_reaction_loads()` and `b.solve_slope_deflection()` are mandatory before calling the max function, this should probably be mentioned in the docstrings